common/OpTestUtil.py

```diff
@@ -1913,6 +1913,54 @@ def wait_for(self, func, timeout, first=0.0, step=1.0, text=None, args=None, kwa
             time.sleep(step)
         return None
 
+    def gather_os_logs(self, list_of_files=[], list_of_commands=[], output_dirname=None):
+        host = self.conf.host()
+        if not output_dirname:
+            output_dirname = "Logs_%s" % (time.asctime(time.localtime())).replace(" ", "_")
+        output_dir = os.path.join(host.results_dir, output_dirname)
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+        self.gather_files_log(list_of_files, output_dir)
+        self.gather_command_log(list_of_commands, output_dir)
+
+    def gather_files_log(self, list_of_files=[], output_dir=None):
+        if not output_dir:
+            raise OpTestError("Provide path to the output directory (output_dir).")
```

Comment:
why not  Handel  in default case as if user not provided we can populate default directory  so we can avoid unnecessary  breakage of utils

Reply:
even I say please have gather_files_log and gather_command_log can be a private function to avoid calling directly

Reply:
I have made the functions private and gotten rid of this check. 
The gather_files_log() and gather_command_log() functions are only being called from inside gather_os_logs(). And we are defining the directory inside the gather_os_logs() function, and passing it as an argument to the gather_file_log() and gather_command_log() functions. Thus, the output_dir won't be a None value in this scenario. 
Is that fine?

Reply:
yes looks good